instrumentation/opentelemetry-instrumentation-botocore/tests/test_botocore_instrumentation.py

```diff
@@ -66,20 +61,17 @@ def test_traced_client(self):
         assert spans
         span = spans[0]
         self.assertEqual(len(spans), 1)
-        self.assertEqual(span.attributes["aws.agent"], "botocore")
-        self.assertEqual(span.attributes["aws.region"], "us-west-2")
-        self.assertEqual(span.attributes["aws.operation"], "DescribeInstances")
-        assert_span_http_status_code(span, 200)
         self.assertEqual(
-            span.resource,
-            Resource(
-                attributes={
-                    "endpoint": "ec2",
-                    "operation": "describeinstances",
-                }
-            ),
+            span.attributes,
+            {
+                "aws.operation": "DescribeInstances",
+                "aws.region": "us-west-2",
+                "aws.request_id": "fdcdcab1-ae5c-489e-9c33-4637c5dda355",
```

Comment:
Not sure why the IDs are sometimes consistent and sometimes not with `moto`.

Something the ID is consistent like this EC2 call:

```
{
    'Reservations': [],
    'ResponseMetadata': {'RequestId': 'fdcdcab1-ae5c-489e-9c33-4637c5dda355',
    'HTTPStatusCode': 200,
    'HTTPHeaders': {'server': 'amazon.com'},
    'RetryAttempts': 0}
}
```

and other times it's randomly generated like in this SQS call:
```
{
    'ResponseMetadata': {
        'RequestId': 'JWH6CYKFXYT2KXQSAYY50T6FK558XMQ7U9SOWLJ5YEQZOKULQ012',
        'HTTPStatusCode': 200,
        'HTTPHeaders': {'server': 'amazon.com',
        'x-amzn-requestid': 'JWH6CYKFXYT2KXQSAYY50T6FK558XMQ7U9SOWLJ5YEQZOKULQ012',
        'x-amz-crc32': '1416772655'},
        'RetryAttempts': 0
    }
}

```

Reply:
Yes! I looked into it, and found this code used to generate the request ID in the `moto` repository:

https://github.com/spulec/moto/blob/9db62d32bf70e18f315305b7915d199e3ba1210a/moto/core/utils.py#L238-L244

It was [explained to me](https://github.com/spulec/moto/issues/3442) that this function is used by _some_ of the resources to attach a request ID.

Notably the `amzn_request_id()` decorator is used on SQS and DynamoDB and that's why their request IDs look like this 🙂 

And some services have hardcoded requests like EC2 `DescribeInstances` (https://github.com/spulec/moto/blob/50d71eccbee12ce5cf2c1ecf61bfcbd60317976d/moto/ec2/responses/instances.py#L479)

I'm not sure why it doesn't format the request IDs in the expected way (lower case joined with hyphens) because it [used to be correct](https://github.com/spulec/moto/blame/2bb3e841d12c5f614971492d241629794f35fe85/moto/core/utils.py#L68).

In that case, should I remove the regex check and just make it a check for the key? Let me know what you think!